pkg/agent/secondarynetwork/podwatch/controller_test.go

```diff
@@ -609,15 +626,65 @@ func TestConfigurePodSecondaryNetwork(t *testing.T) {
 					interfaceName,
 					1500,
 					&ipam.IPAMResult{},
+					nil,
 				).Return(errors.New("interface creation failure"))
 			},
 			expectedErr: "interface creation failure",
 		},
+		{
+			name:        "static MAC address configured",
+			networkType: vlanNetworkType,
+			mtu:         1600,
+			vlan:        101,
+			expectedCalls: func(mockIPAM *podwatchtesting.MockIPAMAllocator, mockIC *podwatchtesting.MockInterfaceConfigurator) {
+				expectedMAC, _ := net.ParseMAC("02:42:ac:11:00:02")
+				mockIPAM.EXPECT().SecondaryNetworkAllocate(podOwner, gomock.Any()).Return(testIPAMResult("148.14.24.100/24", 0), nil)
+				mockIC.EXPECT().ConfigureVLANSecondaryInterface(
+					podName,
+					testNamespace,
+					containerID,
+					containerNetNs(containerID),
+					interfaceName,
+					1600,
+					testIPAMResult("148.14.24.100/24", 101),
+					expectedMAC,
+				).DoAndReturn(func(podName, podNamespace, containerID, netNS, ifName string, mtu int, result *ipam.IPAMResult, mac net.HardwareAddr) error {
+					assert.Equal(t, expectedMAC, mac)
+					return nil
+				})
+			},
+			expectedNetworkStatusAnnot: `[{
+    "name": "net",
+    "ips": [
+        "148.14.24.100"
+    ],
+    "dns": {}
+}]`,
+		},
+		{
+			name:        "invalid MAC address fails",
+			networkType: vlanNetworkType,
+			mtu:         1600,
+			vlan:        101,
+			expectedCalls: func(mockIPAM *podwatchtesting.MockIPAMAllocator, mockIC *podwatchtesting.MockInterfaceConfigurator) {
+				// Expect no call to interface configurator
+			},
+			expectedErr: "",
+		},
 	}
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			pod, cniInfo := testPod(podName, containerID, podIP, element1)
+			var element netdefv1.NetworkSelectionElement
+			switch tc.name {
+			case "invalid MAC address fails":
+				element = element3
+			case "static MAC address configured":
+				element = element2
+			default:
+				element = element1
+			}
```

Comment:
this is not a good way to write table-driven tests in general. Instead, please add `element` as a field in the test case definition.